src/WebJobs.Script/Config/ScriptHostRecycleOptionsSetup.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Azure.WebJobs.Script.Workers.Http;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.Configuration
+{
+    internal sealed class ScriptHostRecycleOptionsSetup : IConfigureOptions<ScriptHostRecycleOptions>
+    {
+        private readonly IOptions<HttpWorkerOptions> _httpWorkerOptions;
+
+        public ScriptHostRecycleOptionsSetup(IOptions<HttpWorkerOptions> httpWorkerOptions)
+        {
+            _httpWorkerOptions = httpWorkerOptions ?? throw new ArgumentNullException(nameof(httpWorkerOptions));
+        }
+
+        public void Configure(ScriptHostRecycleOptions options)
+        {
+            // Enforcing sequential host restarts when a user-specified custom handler port is configured to prevent multiple processes from attempting to bind to the same port concurrently.
+            options.SequentialHostRestartRequired = _httpWorkerOptions.Value?.IsPortManuallySet ?? false;
```

Comment:
This will set it to `false` if someone else previously set to `true`. While I know we don't have anything that _would_ set it to true right now, a more future-proof approach would be:

(also, you can assume `.Value` is non-null.
```suggestion
            if (_httpWorkerOptions.Value.IsPortManuallySet)
            {
                // If statement to ensure we only set this to TRUE and never to FALSE.
                options.SequentialHostRestartRequired = true;
            }
```

this way the logic will only ever toggle sequential restart ON and never OFF